types/scene/wlr_scene.c

```diff
@@ -1027,6 +1041,37 @@ void wlr_scene_shadow_set_color(struct wlr_scene_shadow *shadow, const float col
 	scene_node_update(&shadow->node, NULL);
 }
 
+void wlr_scene_shadow_set_reference_buffer(struct wlr_scene_shadow *shadow,
+		struct wlr_scene_buffer *ref_buffer) {
+	if (shadow->type != WLR_SCENE_SHADOW_TYPE_DROP) {
+		wlr_log(WLR_ERROR, "Trying to set a reference buffer to a shadow with type of box."
+				"Please change its type before calling this function");
+		return;
+	}
+
+	linked_node_init_link(&shadow->buffer_link, &ref_buffer->drop_shadow_link);
+
+	scene_node_update(&shadow->node, NULL);
+}
+
+int wlr_scene_shadow_get_offset(struct wlr_scene_shadow *shadow) {
+	switch (shadow->type) {
+	case WLR_SCENE_SHADOW_TYPE_DROP:
+		return drop_shadow_calc_size(shadow->blur_sigma);
+	default:
+		return shadow->blur_sigma;
+	}
+}
```

Comment:
have you tested that the shadows are functionally identical between box and drop? We'd want the same appearance minus the transparent areas between types. Also, why do you refer to the sigma as offset for the drop shadow?